### DISCUSSION TITLE
Merge event handler when providing multiple targets for x-bind

### DISCUSSION THREAD

**Discussion by simonferndriger** [alpinejs/alpine#3884] [3. Feature ideas]
If it does not mean a huge overhead, it would be nice to allow multiple `x-bind`. Use case is especially for buttons, where you want to bind multiple functionalities to it. For example: `<button x-bind="$accordionButton" x-bind="$loadInfoAboutSomething">`.

**Comment by SimoTod** [comment]
Unfortunately, you can't have 2 HTML attributes with the exact same name. It's a browser thing, not Alpine.

**Comment by simonferndriger** [comment]
Ah OK, well, how about `x-bind="['$accordionButton', '$loadInfoAboutSomething']"`?

**Reply by ekwoka** [reply]
`x-bind="{...accordion, ...button}"`

**Reply by simonferndriger** [reply]
Ah, converting the first one into a new object and then adding as many as needed after. Genius! Thanks :)

**Reply by ekwoka** [reply]
looks like I messed up. I mean spread them all. You need a flat object.

**Reply by simonferndriger** [reply]
Ah OK, but what happens when both `accordion` and `button` have defined a `@click` event handler?

**Reply by ekwoka** [reply]
The second will overwrite.

You'd need to get more complex with a merger if you want to be able to have them combined.

**Reply by simonferndriger** [reply]
OK, I see. So in that case - that complex solution would be my feature request 😊

**Reply by simonferndriger** [reply]
But this is for sure a nice-to-have, not a need-to-have. So if the downside is big, I totally understand if this idea gets discarded.

**Reply by ekwoka** [reply]
Generally, this should just be handled with directives. They'd be quite a lot easier to do, and simple to adapt in userland.
